File: 0201-0300/0246-Strobogrammatic Number/0246-Strobogrammatic Number.cpp
class Solution {
public:
    bool isStrobogrammatic(string num) {
        int n = num.size();
        if (n == 0) {
            return true;
        }
        
        int start = 0, end = n - 1;
        while (start <= end) {
            if (convert(num[start]) != num[end]) {
                return false;
            }
            ++start;
            --end;
        }
        
        return true;
    }
    
private:
    char convert(char ch) {
        if (ch == '0' || ch == '1' || ch == '8') {
            return ch;
        }
        else if (ch == '6') {
            return '9';
        }
        else if (ch == '9') {
            return '6';
        }
        else {
            return ' ';
        }
    }
};

File: 0201-0300/0246-Strobogrammatic Number/0246-Strobogrammatic Number.py
class Solution:
    def isStrobogrammatic(self, num: str) -> bool:
        table = {'0':'0', '1':'1', '8':'8', '6':'9', '9':'6'}
        start, end = 0, len(num) - 1
        while start <= end:
            if num[start] not in table or table[num[start]] != num[end]:
                return False
            start += 1
            end -= 1
        return True
